beacon-chain/blockchain/core.go

```diff
@@ -490,22 +520,22 @@ func (b *BeaconChain) validatorsByHeightShard() ([]*beaconCommittee, error) {
 	return committees, nil
 }
 
-// GetIndicesForHeight returns the attester set of a given height.
-func (b *BeaconChain) GetIndicesForHeight(height uint64) (*pb.ShardAndCommitteeArray, error) {
-	lcs := b.CrystallizedState().LastStateRecalc()
+// getIndicesForSlot returns the attester set of a given height.
+func (b *BeaconChain) getIndicesForHeight(crystallized *types.CrystallizedState, height uint64) (*pb.ShardAndCommitteeArray, error) {
+	lcs := crystallized.LastStateRecalc()
 	if !(lcs <= height && height < lcs+params.CycleLength*2) {
 		return nil, fmt.Errorf("can not return attester set of given height, input height %v has to be in between %v and %v", height, lcs, lcs+params.CycleLength*2)
 	}
-	return b.CrystallizedState().IndicesForHeights()[height-lcs], nil
+	return crystallized.IndicesForHeights()[height-lcs], nil
 }
 
-// GetBlockHash returns the block hash of a given height.
-func (b *BeaconChain) GetBlockHash(slot, height uint64) ([]byte, error) {
+// getBlockHash returns the block hash of a given height.
+func (b *BeaconChain) getBlockHash(active *types.ActiveState, slot, height uint64) ([]byte, error) {
```

Comment:
This pure method can go into another package or file since it doesn’t depend on any of the beacon chain properties. 

This core.go file is getting pretty wild